packages/opentelemetry-core/src/trace/Span.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { randomSpanId, randomTraceId } from '../platform';
+import { isValid } from './spancontext-utils';
+
+/**
+ * This class represents a span.
+ */
+export class Span implements types.Span {
+  private _traceId: string;
+  private _spanId: string;
+  private _parentId?: string;
+  private _traceState?: types.TraceState;
+  private _name: string;
+  private _kind: types.SpanKind;
+  private _attributes: types.Attributes = {};
+  private _links: types.Link[] = [];
+  private _events: types.Event[] = [];
+  private _status: types.Status = {
+    code: types.CanonicalCode.OK,
+  };
+  private _ended = false;
+  private _startTime: number;
+  private _duration = 0;
+
+  /** Constructs a new Span instance. */
+  constructor(
+    readonly _parentTracer: types.Tracer,
+    readonly _spanName: string,
```

Comment:
Could you make this `private readonly _spanName: string,` to avoid needing the separate `this._name = _spanName`?

Not sure whether it's better called `_name` or `_spanName`

Reply:
`readonly` is not possible as we have `updateName` [method](https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-types/src/trace/span.ts#L91) to update the span name.

Reply:
Ah, ok. Can you say then `private _spanName` so that you don't need a separate `_name` variable?